modules/sharedIdSystem.js

```diff
@@ -1,373 +1,107 @@
+   * @param {SubmoduleConfig} config
+   * @returns {{pubcid:string}}
    */
-  decode(value) {
-    return (value) ? encodeId(value) : undefined;
+  decode(value, config) {
+    return {'pubcid': value};
   },
-
   /**
-   * performs action to obtain id and return a value.
+   * performs action to obtain id
    * @function
-   * @param {SubmoduleConfig} [config]
-   * @param {ConsentData|undefined} consentData
-   * @returns {sharedId}
+   * @param {SubmoduleConfig} [config] Config object with params and storage properties
+   * @param {Object} consentData
+   * @param {string} storedId Existing pubcommon id
+   * @returns {IdResponse}
    */
-  getId(config, consentData) {
-    const coppa = coppaDataHandler.getCoppa();
-    if (coppa) {
-      utils.logInfo('SharedId: IDs not provided for coppa requests, exiting SharedId');
-      return;
+  getId: function (config = {}, consentData, storedId) {
+    const {params: {create = true, pixelUrl} = {}} = config;
+    let newId = storedId;
+    if (!newId) {
+      try {
+        if (typeof window[PUB_COMMON_ID] === 'object') {
+          // If the page includes its own pubcid module, then save a copy of id.
+          newId = window[PUB_COMMON_ID].getId();
+        }
+      } catch (e) {
+      }
+
+      if (!newId) newId = (create && utils.hasDeviceAccess()) ? utils.generateUUID() : undefined;
     }
-    const resp = function (callback) {
-      utils.logInfo('SharedId: Sharedid doesnt exists, new cookie creation');
-      ajax(sharedIdUrl(consentData), idGenerationCallback(callback), undefined, {method: 'GET', withCredentials: true});
-    };
-    return {callback: resp};
-  },
 
+    const pixelCallback = this.makeCallback(pixelUrl, newId);
+
+    return {id: newId, callback: pixelCallback};
+  },
   /**
-   * performs actions even if the id exists and returns a value
-   * @param config
-   * @param consentData
-   * @param storedId
-   * @returns {{callback: *}}
+   * performs action to extend an id.  There are generally two ways to extend the expiration time
+   * of stored id: using pixelUrl or return the id and let main user id module write it again with
+   * the new expiration time.
+   *
+   * PixelUrl, if defined, should point back to a first party domain endpoint.  On the server
+   * side, there is either a plugin, or customized logic to read and write back the pubcid cookie.
+   * The extendId function itself should return only the callback, and not the id itself to avoid
+   * having the script-side overwriting server-side.  This applies to  pubcid.
+   *
+   * On the other hand, if there is no pixelUrl, then the extendId should return storedId so that
+   * its expiration time is updated.  Sharedid, however, will have to be updated by this submodule
+   * separately.
+   *
+   * @function
+   * @param {SubmoduleParams} [config]
+   * @param {ConsentData|undefined} consentData
+   * @param {Object} storedId existing id
+   * @returns {IdResponse|undefined}
```

Comment:
Looks like the extendId function comments are included but the pubCommonId function itself was not placed in here. Was this intentional? The domainOverride function was also not included as it was in the last PR

Reply:
No, that wasn`t intentional, will remove it

Reply:
updated